Makefile

```diff
@@ -130,53 +135,87 @@ dockerbuild-go:
 .PHONY: dockerbuild-go
 
+dockerbuild-test:
+	docker build -f DockerFileTests --build-arg GCLOUD_KEY="$(GCLOUD_KEY)" \
+	             --build-arg PROJECT_NAME="$(PROJECT_NAME)" \
+	             --build-arg CLUSTER_NAME="$(CLUSTER_NAME)" \
+	             --build-arg CLUSTER_ZONE="$(CLUSTER_ZONE)" \
+	             -t go-spacemesh-python:$(BRANCH) .
+.PHONY: dockerbuild-test
+
 dockerpush: dockerbuild-go
 	echo "$(DOCKER_PASSWORD)" | docker login -u "$(DOCKER_USERNAME)" --password-stdin
 	docker tag $(DOCKER_IMAGE_REPO):$(BRANCH) spacemeshos/$(DOCKER_IMAGE_REPO):$(BRANCH)
 	docker push spacemeshos/$(DOCKER_IMAGE_REPO):$(BRANCH)
+
 ifeq ($(BRANCH),develop)
 	docker tag $(DOCKER_IMAGE_REPO):$(BRANCH) spacemeshos/$(DOCKER_IMAGE_REPO):$(SHA)
 	docker push spacemeshos/$(DOCKER_IMAGE_REPO):$(SHA)
 endif
 .PHONY: dockerpush
 
-dockerbuild-test:
-	docker build -f DockerFileTests --build-arg GCLOUD_KEY="$(GCLOUD_KEY)" \
-	             --build-arg PROJECT_NAME="$(PROJECT_NAME)" \
-	             --build-arg CLUSTER_NAME="$(CLUSTER_NAME)" \
-	             --build-arg CLUSTER_ZONE="$(CLUSTER_ZONE)" \
-	             -t go-spacemesh-python:latest .
-.PHONY: dockerbuild-test
 
 ifdef TEST
 DELIM=::
 endif
 
-dockerrun-test: dockerbuild-test
+
+dockerrun-p2p:
+ifndef ES_PASSWD
+	$(error ES_PASSWD is not set)
+endif
+	docker run --rm -e ES_PASSWD="$(ES_PASSWD)" \
+		-e GOOGLE_APPLICATION_CREDENTIALS=./spacemesh.json \
+		-e CLIENT_DOCKER_IMAGE="spacemeshos/$(DOCKER_IMAGE_REPO):$(BRANCH)" \
+		-it go-spacemesh-python:$(BRANCH) pytest -s p2p/test_p2p.py --tc-file=p2p/config.yaml --tc-format=yaml
+.PHONY: dockerrun-p2p
+
+dockertest-p2p: dockerbuild-test dockerrun-p2p
+.PHONY: dockertest-p2p
+
+dockerrun-mining:
 ifndef ES_PASSWD
 	$(error ES_PASSWD is not set)
 endif
+	docker run --rm -e ES_PASSWD="$(ES_PASSWD)" \
+		-e GOOGLE_APPLICATION_CREDENTIALS=./spacemesh.json \
+		-e CLIENT_DOCKER_IMAGE="spacemeshos/$(DOCKER_IMAGE_REPO):$(BRANCH)" \
+		-it go-spacemesh-python:$(BRANCH) pytest -s test_bs.py --tc-file=config.yaml --tc-format=yaml
+.PHONY: dockerrun-mining
 
-	docker run -e ES_PASSWD="$(ES_PASSWD)" \
-             -e GOOGLE_APPLICATION_CREDENTIALS=./spacemesh.json \
-             -e CLIENT_DOCKER_IMAGE="spacemeshos/$(DOCKER_IMAGE_REPO):$(BRANCH)" \
-             -it go-spacemesh-python pytest -s p2p/test_p2p.py --tc-file=p2p/config.yaml --tc-format=yaml
+dockertest-mining: dockerbuild-test dockerrun-mining
+.PHONY: dockertest-mining
+
+dockerrun-hare:
+ifndef ES_PASSWD
+	$(error ES_PASSWD is not set)
+endif
+	docker run --rm -e ES_PASSWD="$(ES_PASSWD)" \
+		-e GOOGLE_APPLICATION_CREDENTIALS=./spacemesh.json \
+		-e CLIENT_DOCKER_IMAGE="spacemeshos/$(DOCKER_IMAGE_REPO):$(BRANCH)" \
+		-it go-spacemesh-python:$(BRANCH) pytest -s hare/test_hare.py --tc-file=hare/config.yaml --tc-format=yaml
+.PHONY: dockerrun-hare
+
+dockertest-hare: dockerbuild-test dockerrun-hare
+.PHONY: dockertest-hare
+
+dockerrun-sync:
+ifndef ES_PASSWD
```

Comment:
maybe we can do this in a more general context instead of in every cmd

Reply:
I do this in order to be able to run each command by it's own